docs/source/torchao_vllm_integration.md

```diff
@@ -0,0 +1,398 @@
+```
+
+(register-with-torchao)=
+#### 3. Register with TorchAO's Quantization System
+
+```python
+from torchao.quantization.transform_module import register_quantize_module_handler
+
+@register_quantize_module_handler(MyNewQuantConfig)
+def _my_quant_transform(module: torch.nn.Module, config: MyNewQuantConfig):
+    """Transform function that applies your quantization to a module"""
+    weight = module.weight
+
+    # Your quantization logic here
+    quantized_weight = my_quantization_function(weight, config)
+
+    # Replace the weight with your quantized tensor
+    module.weight = torch.nn.Parameter(quantized_weight, requires_grad=False)
+    return module
+```
+
+```{important}
+The {func}`torchao.quantization.transform_module.register_quantize_module_handler` decorator registers your config class with TorchAO's quantization system.
+```
+
+(key-implementation-details)=
+### Key Implementation Details
+
+(hardware-specific-linear-ops)=
+#### Hardware-Specific Linear Operations
+Your quantized tensor's forward pass determines hardware support and what actually gets called when {func}`torch.nn.functional.linear` is called.
+
+```python
+@MyQuantizedTensor.implements(torch.nn.functional.linear)
+def _(func, types, args, kwargs):
+    input_tensor, weight_tensor, bias = args[0], args[1], args[2] if len(args) > 2 else None
+
+    # This is where you define what hardware your method supports
+    if hasattr(weight_tensor, 'use_cutlass_kernel'):
+        return my_cutlass_linear(input_tensor, weight_tensor, bias)
+    elif hasattr(weight_tensor, 'use_triton_kernel'):
+        return my_triton_linear(input_tensor, weight_tensor, bias)
+    else:
+        # Fallback - dequantize and use standard linear
+        return torch.nn.functional.linear(
+            input_tensor, weight_tensor.dequantize(), bias
+        )
+```
+
+(compilation-benefits)=
+#### Compilation Benefits
+The overhead of tensor subclasses disappears with {func}`torch.compile`, this is on by default in VLLM.
+
+(trade-off-tensor-subclasses)=
+### Trade Off of Tensor Subclasses
+1. **Compilation**: is essential for removing subclass overhead. Without it unless your model is extremely gpu bound the overhead of dispatch on the CPU can severely impact performance.
+2. The checkpoint defines the behavior of the model. You might be saying "don't all checkpoints do this". This is true, however people typically solely think of a torch.Tensor as its data. When in actuality its a true class where it brings the Dispatcher and all the kernels ATen has registered to it. When you define your tensor subclass, you are building a separate little world. One w/ a different representation of data, but also one where you need to explicitly define what ops you support and have implementations for all the hardware you want to support. This can feel a little like spooky action at a distance at first. But it can be very powerful. Case in point is being able to support TP with only 3 definitions.
```

Comment:
Link to this related section here? https://docs.pytorch.org/ao/stable/subclass_basic.html#what-are-tensor-subclasses. "For more details, please refer to ..."